plugins/parcelport/verbs/connection_handler.hpp

```
@@ -0,0 +1,92 @@
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_PARCELSET_POLICIES_CONNECTION_HANDLER_HPP
+#define HPX_PARCELSET_POLICIES_CONNECTION_HANDLER_HPP
+
+#include <hpx/config/warnings_prefix.hpp>
+#include <hpx/runtime/parcelset/locality.hpp>
+#include <hpx/runtime/parcelset/parcelport_impl.hpp>
+//
+#include <plugins/parcelport/verbs/rdma/rdma_logging.hpp>
+#include <plugins/parcelport/verbs/rdma/rdma_error.hpp>
+#include <plugins/parcelport/verbs/rdma/verbs_device.hpp>
+//
+#include <sstream>
+#include <type_traits>
+
+namespace hpx {
+namespace parcelset {
+namespace policies {
+namespace verbs
+{
+    class HPX_EXPORT parcelport;
+}}
+
+    template <>
+    struct connection_handler_traits<policies::verbs::parcelport>
+    {
+        typedef policies::verbs::sender_connection      connection_type;
+        typedef HPX_PARCELPORT_VERBS_HAVE_BOOTSTRAPPING send_early_parcel;
+        typedef std::true_type                          do_background_work;
+        typedef std::true_type                          send_immediate_parcels;
+
+        static const char * type()
+        {
+            return "verbs";
+        }
+
+        static const char * pool_name()
+        {
+            return "parcel-pool-verbs";
+        }
+
+        static const char * pool_name_postfix()
+        {
+            return "-verbs";
+        }
+    };
+
+namespace policies {
+namespace verbs
+{
+        uint32_t get_verbs_device_address(const char *devicename, const char *iface,
+            char *hostname)
+        {
+          FUNC_START_DEBUG_MSG
+          // Find the address of the I/O link device.
+          verbs_device_ptr linkDevice;
+          try {
+            linkDevice = verbs_device_ptr(new verbs_device(devicename, iface));
+          }
+          catch (rdma_error& e) {
+            LOG_ERROR_MSG("error opening InfiniBand device: " << e.what());
```

I'm not sure what to do about errors in the parcelport in general. The error handling is at best 'inconsistent' - in some places I throw, and in others I just log an error and return a fail. 
Here's we should probably rethrow the error and allow the app to terminate, or be handled above. 
An #issue to make the error handling consistent would be a good idea.